### PR TITLE
Minor repo-updater code cleanup

### PR DESCRIPTION
**Pull request by eseliger**
Just some minor code cleanup from reading through the init code of repo-updater. Also finally removed some more global assignments from the enterprise split. 

## Test plan

sg start still works and CI passes.

### CODE REVIEW

File: internal/repos/scheduler.go [hunk 1]
```diff
@@ -158,6 +117,10 @@ func (s *UpdateScheduler) runScheduleLoop(ctx context.Context) {
 			return
 		}
 
+		if conf.Get().DisableAutoGitUpdates {
+			continue
```
Comment (eseliger) [inline internal/repos/scheduler.go:121]: this feels much simpler than the above logic that cancels the context and restarts the entire routine.

File: internal/repos/scheduler.go [hunk 2]
```diff
@@ -148,6 +91,22 @@ func NewUpdateScheduler(logger log.Logger, db database.DB) *UpdateScheduler {
 	}
 }
 
+func (s *UpdateScheduler) Start() {
```
Comment (eseliger) [inline internal/repos/scheduler.go:94]: Made it a goroutine.BackgroundRoutine so we get the same monitoring as we have for other routines. Ideally, this wouldn't spawn 2 untracked goroutines, but we can leave that for later.

File: cmd/repo-updater/shared/main.go [hunk 3]
```diff
@@ -111,96 +106,71 @@ func Main(ctx context.Context, observationCtx *observation.Context, ready servic
...
-		Sourcer: src,
-		Store:   store,
-		// We always want to listen on the Synced channel since external service syncing
-		// happens on both Cloud and non Cloud instances.
-		Synced:  make(chan repos.Diff),
-		Now:     clock,
-		ObsvCtx: observation.ContextWithLogger(logger.Scoped("syncer", "repo syncer"), observationCtx),
-	}
-
-	server.Syncer = syncer
```
Comment (pjlast) [inline cmd/repo-updater/shared/main.go:153]: Hmmmm I recall this being done after the `server.RateLimitSyncer.SyncRateLimiters` call for a reason, but can't quite recall why right now

File: internal/repos/scheduler.go (hunk 2 above)
Comment (pjlast) [reply internal/repos/scheduler.go:94]: Yet the tests pass and I don't see any issues locally, so 🤷‍♂️

File: cmd/repo-updater/shared/main.go (hunk 3 above)
Comment (pjlast) [reply cmd/repo-updater/shared/main.go:153]: Yet the tests pass and I don't see any issues locally so 🤷‍♂️

Comment (pjlast) [reply cmd/repo-updater/shared/main.go:153]: Okay, the change I made was from here: https://github.com/sourcegraph/sourcegraph/pull/46504

Which required the syncer to be set earlier. So my above message is incorrect.

The correct thing would have been to do what you did here, so disregard these messages :)

File: internal/repos/scheduler.go (hunk 1 above)
Comment (pjlast) [reply internal/repos/scheduler.go:121]: Yep, this is much simpler 😅

**Review by pjlast — APPROVED** [review]

**Pull request merged**